File: backbone/app/contracts/api/accounts.py
from pydantic import BaseModel, ConfigDict, Field


class AccountSummary(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    tenant_id: str
    name: str
    domain: str | None = None
    lifecycle_stage: str


class ListAccountsQuery(BaseModel):
    model_config = ConfigDict(extra="forbid")

    limit: int = Field(default=50, ge=1, le=200)
    offset: int = Field(default=0, ge=0)


class ListAccountsResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[AccountSummary]
    count: int

File: ui/services/accounts.ts
import { buildUrl } from './api'

export type AccountSummary = { id: string; tenant_id: string; name: string; domain?: string; lifecycle_stage?: string }

export async function fetchAccounts({ limit = 50, offset = 0 }: { limit?: number; offset?: number }) {
  const url = buildUrl('/accounts', { limit, offset })
  const res = await fetch(url)
  if (!res.ok) throw new Error(`API error ${res.status}`)
  const payload = await res.json()
  // payload is SuccessEnvelope { success: true, data: { items, count } }
  return payload.data
}
